stetho-sample/src/debug/java/com/facebook/stetho/sample/CalendarProviderDatabasePeer.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.sample;
+
+import android.annotation.TargetApi;
+import android.content.ContentResolver;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteException;
+import android.net.Uri;
+import android.os.Build;
+import android.provider.CalendarContract;
+import com.facebook.stetho.inspector.jsonrpc.JsonRpcPeer;
+import com.facebook.stetho.inspector.protocol.module.Database;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is example of extending database domain by adding calendar provider
+ */
+public class CalendarProviderDatabasePeer extends Database.DatabasePeer {
+
+  private final static String sDatabaseName = "provider-calendars";
+
+  public CalendarProviderDatabasePeer(Context context) {
+    super(context);
+  }
+
+  @Override
+  protected void onRegistered(JsonRpcPeer peer) {
+    Database.DatabaseObject databaseParams = new Database.DatabaseObject();
```

Comment:
This junk should not be leaked into the implementor.  Strongly prefer a simpler mechanism of listing the databases.  The timing and behaviour of this should be managed more internally to the Database module and/or the actual peer manager.